Adds the whats new page for 8.14 (#183556)
elastic/kibana: docs/user/whats-new.asciidoc

[discrete]
==== Transitioning {esql} from Tech Preview to General Availability!

{esql} offers a streamlined way to filter, transform, and analyze data in {es}. Its intuitive design, utilizing "pipes" (|) for step-by-step data exploration, enables you to easily compose powerful queries for detailed analysis. Whether you're a developer, SRE, or Security Analyst, {esql} empowers you to uncover specific events, perform robust statistical analyses, and create compelling visualizations. As we move from tech preview to general availability, discover the enhanced capabilities of {esql} and elevate your data operations.

[discrete]
==== {esql} Query History

We've enhanced the {esql} editor to improve your workflow. You can now view and re-run your last 20 {esql} queries directly within Discover, {esql} charts, alerts, and maps. This feature makes it easier to manage your queries and streamline your tasks. 

[role="screenshot"]
image::images/query-history-in-discover.png[An image of an ES|QL query history in Discover.]

[role="screenshot"]
image::images/query-history-in-dashboard.png[An image of the auto option.]

[discrete]
==== Document comparison mode in Discover & {esql}

> We are enabling users to select and compare documents or fields. This functionality will be a game-changer for tasks like debugging, allowing you to perform detailed comparisons, such as diffing SIP messages of a certain ID across multiple documents in {es}. This will streamline your analysis and troubleshooting processes.

Reviewer: ```suggestion
You can now select and compare documents or fields. This functionality streamlines analysis and troubleshooting tasks by allowing you to perform detailed comparisons, such as diffing SIP messages of a certain ID across multiple documents in {es}.
```
